src/EventStore.Core/DataStructures/ProbabilisticFilter/AlignedMemory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace EventStore.Core.DataStructures.ProbabilisticFilter {
+	// net6 can do aligned allocations directly
+	// but we need to overallocate and do our own alignment.
+	public sealed unsafe class AlignedMemory : IDisposable {
+		private readonly IntPtr _intPtr;
+		private readonly long _size;
+		private readonly long _bytesAllocated;
+		private bool _disposed;
+
+		/// contents of memory are not initialized
+		public AlignedMemory(long size, int alignTo) : this(new IntPtr(size), alignTo) {
+		}
+
+		/// contents of memory are not initialized
+		public AlignedMemory(IntPtr size, int alignTo) {
+			_size = (long)size;
+			var bytesToAllocate = size + alignTo;
+			_intPtr = Marshal.AllocHGlobal(bytesToAllocate);
+			_bytesAllocated = (long)bytesToAllocate;
+			GC.AddMemoryPressure(_bytesAllocated);
```

Comment:
I dont think this is actually needed.

Also why is size being passed as an IntPtr then converted to a long?! This is just ODD!